### PR TITLE
[ENH] Parallelize ForecastingBenchmark

### LINKED ISSUE #6559: [ENH] Enable parallel execution of ForecastingBenchmarks
**Is your feature request related to a problem? Please describe.**
Currently, in the ForecastingBenchmark each forecaster on each cv fold is evaluated separately. I would propose to enable options for parallel execution to fasten it up.

**Describe the solution you'd like**
When initializing the benchmarking, one can pass options for parallelization that are then provided to the `evaluate method`.

### PR DESCRIPTION
**Pull request by benHeid**
#### Reference Issues/PRs
fixes #6559

#### What does this implement/fix? Explain your changes.
add backend options to benchmarking and passes them to evaluate

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

File: sktime/benchmarking/forecasting.py [hunk 1]
```diff
@@ -28,14 +30,46 @@ def forecasting_validation(
...
-
+    backend : {"dask", "loky", "multiprocessing", "threading"}, by default None.
+        Runs parallel evaluate for each task if specified.
+
+        - "None": executes loop sequentally, simple list comprehension
+        - "loky", "multiprocessing" and "threading": uses ``joblib.Parallel`` loops
+        - "joblib": custom and 3rd party ``joblib`` backends, e.g., ``spark``
+        - "dask": uses ``dask``, requires ``dask`` package in environment
+        - "dask_lazy": same as "dask",
+        but changes the return to (lazy) ``dask.dataframe.DataFrame``.
```
Comment (fkiraly) [inline sktime/benchmarking/forecasting.py:41]: I think this needs to be indented to render correctly.

Also, will it run the benchmark lazily, or eagerly?

**Review by fkiraly — APPROVED** [review]
Logic-wise straightforward.

I think there are some docstring formatting issues (e.g., indentation upon bullet point) which you may want to check how they render before you merge.

**Review by fkiraly — CHANGES_REQUESTED** [review]
We should:

* add tests covering the various backends. You may want to review and merge https://github.com/sktime/sktime/pull/6566 first to avoid merge conflicts
    * as far as I understand, different backends can be selected for different tasks? We should also ensure this case is covered.
* fix code formatting issues

**Comment by benHeid** [comment]
> We should:
> 
> 
> 
> * add tests covering the various backends. You may want to review and merge https://github.com/sktime/sktime/pull/6566 first to avoid merge conflicts
> 
>     * as far as I understand, different backends can be selected for different tasks? We should also ensure this case is covered.
> 
> * fix code formatting issues

Only one backend for all tasks within one benchmarking object is possible with this PR.

**Comment by fkiraly** [comment]
> Only one backend for all tasks within one benchmarking object is possible with this PR.

Why is it an arg of `_add_task`  then?

**Comment by benHeid** [comment]
> > Only one backend for all tasks within one benchmarking object is possible with this PR.
> 
> Why is it an arg of `_add_task` then?

This was a mistake.

File: sktime/benchmarking/forecasting.py (hunk 1 above)
Comment (benHeid) [reply sktime/benchmarking/forecasting.py:41]: I suppose that the docs are completely missing or? 

I think the benchmarks will be executed if `run` is called.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Looks good.

Silly question, what does this parallelize over? I think the `ForecastingBenchmark` passes single models to `evaluate`. Would it not be better to parallelize at a higher level, namely in what is currently the minikotsu lib´s `run` function? The loops in there are not parallel.

**Comment by benHeid** [comment]
Yes would be possible. However, this might impact the runtime measurements more seriously. In the current case only the same estimator is executed in parallel. In the other case, different estimator are executed in parallel. Thus an estimator that is implemented parallel could be executed together with a single threaded estimator. In this case the parallel executed estimator might impact the runtime measurements of the single threaded one.

File: sktime/benchmarking/forecasting.py (hunk 1 above)
Comment (fkiraly) [reply sktime/benchmarking/forecasting.py:41]: oh, I see, there is indeed no docs page for this! We need to add to the API referenc, but that seems like a separate issue.

Comment (benHeid) [reply sktime/benchmarking/forecasting.py:41]: #6649

**Comment by fkiraly** [comment]
Oh, I see. That is a very subtle point!

**Review by fkiraly — APPROVED** [review]
Is this ready for the release, from your side?

**Comment by benHeid** [comment]
Yes

**Pull request merged**